services/harmony/app/frontends/service-image-tags.ts

```diff
@@ -335,3 +396,74 @@ export async function updateServiceImageTag(
   res.statusCode = 202;
   res.send({ 'tag': tag });
 }
+
+/**
+ * Set the enabled field in service_deployment table to the given vale.
+ * @param value - The boolean value to be set for the enabled field
+ */
+async function setEnabled( value: boolean ): Promise<void> {
+  const sql = `update service_deployment set enabled=${value}, updated_at=CURRENT_TIMESTAMP`;
+  await db.transaction(async (tx) => {
+    await tx.raw(sql);
+  });
+}
+
+/**
+ * Get the current enable/disable state of service image tag update endpoint
+ * @param req - The request object
+ * @param res  - The response object
+ * @param _next  - The next middleware in the chain
+ */
+export async function getServiceImageTagState(
+  req: HarmonyRequest, res: Response, _next: NextFunction,
+): Promise<void> {
+  const enabled = await getEnabled();
+  res.statusCode = 200;
+  res.send({ 'enabled': enabled });
+}
+
+/**
+ * Set the enabled flag of service image tag update endpoint to the given value
+ * @param req - The request object
+ * @param res  - The response object
+ * @param enabled  - The boolean value of enabled
+ */
+export async function setServiceImageTagEnabled(
+  req: HarmonyRequest, res: Response, enabled: boolean,
+): Promise<void> {
+  const validations = [
+    validateUserIsInAdminGroup,
+  ];
+
+  for (const validation of validations) {
+    if (! await validation(req, res)) return;
+  }
+
+  await setEnabled(enabled);
+  res.statusCode = 200;
+  res.send({ 'enabled': enabled });
+}
+
+/**
+ * Enable the service image tag update endpoint
+ * @param req - The request object
+ * @param res  - The response object
+ * @param _next  - The next middleware in the chain
+ */
+export async function enableServiceImageTag(
```

Comment:
I think this and the next function should go away and you should just call `setServiceImageTagEnabled` directly.